app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -123,6 +129,24 @@ def process_collection(collection, key)
         end
       end
 
+      def gather_data_for_this_region(arm_service)
+        results = []
+        @data[:resource_groups].each do |resource_group|
+          results << arm_service.list(resource_group[:name])
```

Comment:
Maybe not something for this PR, but @djberg96 and @bronaghs, I wonder if there's a way for the Armrest API to take on this work and streamline these calls.  Something like, can a service in the Armrest gem make a single call to Azure and return a list of resources given a list of resource groups.